src/wp-includes/rest-api/endpoints/class-wp-rest-themes-controller.php

```diff
@@ -331,6 +332,22 @@ public function prepare_item_for_response( $item, $request ) {
 			$data['is_block_theme'] = $theme->is_block_theme();
 		}
 
+		if ( rest_is_field_included( 'stylesheet_uri', $fields ) ) {
+			if ( $this->is_same_theme( $theme, $current_theme ) ) {
+				$data['stylesheet_uri'] = get_stylesheet_directory_uri();
+			} else {
+				$data['stylesheet_uri'] = $theme->get_stylesheet_directory_uri();
+			}
```

Comment:
Is there a reason you can't use `$theme->get_stylesheet_directory_uri();` in both instances? Do the values differ in any way? 

Same for template so I won't repeat myself.

Reply:
Good question — it was flagged above in the review by @TimothyBJacobs here: https://github.com/WordPress/wordpress-develop/pull/6399#issuecomment-2061349914 — the global function can filter the value for the active theme, so when we match against the current theme, we call the global function in order for that filter to be allowed within the API response.

Here's where the filter lives:

https://github.com/WordPress/wordpress-develop/blob/11bc9c9e8c628bff8a0cca3bcd9eff151164548b/src/wp-includes/theme.php#L239

Reply:
Thanks Andrew, that makes sense.